pkg/operator/status.go

```diff
@@ -374,13 +367,13 @@ func isMachineConfigPoolConfigurationValid(pool *mcfgv1.MachineConfigPool, versi
 		// The bootstrapped MCs fragments have this annotation, however, we don't fail (???) if they don't have
 		// the annotation for some reason.
 		if !ok && pool.Status.Configuration.Name == mcName {
-			return fmt.Errorf("%s must be created by controller version %s", mcName, version)
+			return fmt.Errorf("%s must be created by controller version %s: %v", mcName, version, machineConfigPoolStatus(pool))
 		}
 		// user provided MC fragments do not have the annotation, so we just skip the version check there.
 		// The check below is for: 1) the generated MC for the pool, and 2) the bootstrapped fragments
 		// that do have this annotation set with a version.
 		if ok && v != version {
-			return fmt.Errorf("controller version mismatch for %s expected %s has %s", mcName, version, v)
+			return fmt.Errorf("controller version mismatch for %s expected %s has %s: %v", mcName, version, v, machineConfigPoolStatus(pool))
```

Comment:
Clarification question, you say we're not reporting the error up because the extension field is broken, once that's fixed, does the pool status here report the same error as before?

Reply:
it'll have the additional `, machineConfigPoolStatus(pool)` which is also what we expose in the Extension field - kind of redundant maybe. So right now w/o this fix, it doesn't work because extension wasn't working, now with this fix, besides extension working, we'll always have the pool status in the MCO degraded reason

Reply:
> you say we're not reporting the error up because the extension field is broken,

also, not fully, see #1857 for mcd errors

Reply:
Yeah, ok, I'm fine with the redundancy, I just wanted to check if my understanding is correct, thanks!